pkg/asset/installconfig/gcp/validation.go

```diff
@@ -32,6 +33,22 @@ func validateProject(client API, ic *types.InstallConfig, fieldPath *field.Path)
 		if _, found := projects[ic.GCP.ProjectID]; !found {
 			return append(allErrs, field.Invalid(fieldPath.Child("project"), ic.GCP.ProjectID, "invalid project ID"))
 		}
+
+	}
+	return allErrs
+}
+
+func validatePublicZoneID(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.PublicZoneID != "" {
+		zone, err := client.GetPublicDNSZoneByID(context.TODO(), ic.GCP.ProjectID, ic.GCP.PublicZoneID)
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("publicZoneID"), ic.GCP.PublicZoneID, err.Error()))
```

Comment:
i think we should check for not found error, and that should return validation error that public zone provided was not found.
and for all other errors return https://www.godoc.org/k8s.io/apimachinery/pkg/util/validation/field#InternalError

Reply:
Fixed!